src/components/WeekDayScheduler.js

```diff
@@ -0,0 +1,198 @@
+    );
+
+    setSelectedWDays(sameDays);
+  }, [selectedDay]);
+
+  function handlestartHourChange(e) {
+    const { value } = e.target;
+
+    if (value === "") {
+      setstartHour("");
+      return;
+    }
+
+    if (value <= 0) {
+      setstartHour(0);
+      return;
+    }
+    if (value >= 23) {
+      setstartHour(23);
+      return;
+    }
+
+    if (value >= endHour) {
+      setstartHour(parseInt(value));
+      setEndHour(parseInt(value) + 1);
+      return;
+    }
+
+    setstartHour(parseInt(value));
+  }
+
+  function handleEndHourChange(e) {
+    const { value } = e.target;
+
+    if (value === "") {
+      setEndHour("");
+      return;
+    }
+
+    if (value <= 1) {
+      setEndHour(1);
+      return;
+    }
+    if (value >= 24) {
+      setEndHour(24);
+      return;
+    }
+
+    if (value <= startHour) {
+      setEndHour(startHour + 1);
+      return;
+    }
+
+    setEndHour(parseInt(value));
+  }
+
+  async function handleAdd() {
+    try {
+      if (!setSelectedWDays.length) {
+        return;
+      }
+      for (const exisiting of selectedWDays) {
+        if (startHour < exisiting.startHour && endHour > exisiting.startHour) {
+          setErrorMessage(EXISTED_TIMELINE_WARNING);
+          return;
+        }
+
+        if (
+          startHour >= exisiting.startHour &&
+          startHour <= exisiting.endHour
+        ) {
+          setErrorMessage(EXISTED_TIMELINE_WARNING);
+          return;
+        }
+      }
+
+      dispatch(
+        updateAvailableDates({
+          userId,
+          availableDates: {
+            type: TYPE_WEEKDAY,
+            day: selectedDay,
+            startHour,
+            endHour,
+          },
+        })
+      );
+    } catch (err) {
+      setErrorMessage(err.message);
+    }
+  }
+
+  function handleErrorMessage(error) {
+    setErrorMessage(error);
+  }
+
+  return (
+    <Container>
+      {errorMessage && (
+        <Modal onClick={setErrorMessage} width="50rem" height="20rem">
+          {errorMessage}
+        </Modal>
+      )}
+      <div className="input-fields">
+        <select name="day" onChange={(e) => setSelectedDay(e.target.value)}>
+          <option value="0">일</option>
+          <option value="1">월</option>
+          <option value="2">화</option>
+          <option value="3">수</option>
+          <option value="4">목</option>
+          <option value="5">금</option>
+          <option value="6">토</option>
+        </select>
+        <div>
+          <input
+            className="input"
+            onChange={handlestartHourChange}
+            value={startHour}
+          />
+          <input
+            className="input"
+            onChange={handleEndHourChange}
+            value={endHour}
+          />
+          <button onClick={handleAdd}>저장하기</button>
+        </div>
+      </div>
+
+      {Array.from(Array(7).keys()).map((day) => (
+        <Row key={day} dayNumber={day} onError={handleErrorMessage} />
```

Comment:
```suggestion
        <Row key={day} dayNumber={day} onError={handleErrorMessage} />
```
리팩토링할 때 Row 라는 제너럴한 이름보다 컴포넌트에 대해서 설명할 수 있는 이름으로 하면 좋을 것 같네요!!